kolibri/utils/server.py

```diff
@@ -129,73 +126,31 @@ def port_is_available_on_host(host, port):
     return True
 
 
-class PortCache:
-    def __init__(self):
-        self.values = {}
-        self.occupied_ports = set()
-        self.load()
-
-    def lock_port(self, port):
-        if port:
-            self.occupied_ports.add(port)
-
-    def register_port(self, port):
-        self.values[port] = True
-        self.save()
-
-    def get_port(self, host):
-        if self.values:
-            try:
-                port = next(
-                    p
-                    for p in self.values
-                    if not self.values[p] and p not in self.occupied_ports
-                )
-                if port:
-                    if port_is_available_on_host(host, port):
-                        self.values[port] = True
-                        return port
-            except StopIteration:
-                pass
-        return None
-
-    def save(self):
-        with open(PORT_CACHE, "w") as f:
-            f.write("\n".join(str(p) for p in self.values))
-
-    def load(self):
-        try:
-            with open(PORT_CACHE, "r") as f:
-                for port in f.readlines():
-                    self.values[int(port)] = False
-        except IOError:
-            pass
-
-
-port_cache = PortCache()
-
-
 class ServerPlugin(BaseServerPlugin):
-    def subscribe(self):
-        super(ServerPlugin, self).subscribe()
-        self.bus.subscribe("ENTER", self.ENTER)
+    def __init__(self, *args, **kwargs):
+        httpserver = kwargs.get("httpserver")
+        if not isinstance(httpserver, BaseServer):
+            raise TypeError("httpserver must be a cheroot.wsgi.BaseServer")
+        super(ServerPlugin, self).__init__(*args, **kwargs)
```

Comment:
```suggestion
        # The server is initialized without a bind address before it is passed into the plugin.
        # Because of the property setter below for `bind_addr` the setting of `self.bind_addr`
        # in the super invocation here, results in the httpserver's `bind_addr` property being set.
        super(ServerPlugin, self).__init__(*args, **kwargs)
```